ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -42,6 +42,91 @@ inline bool use_multicore_device_tilize(
     return num_tiles_in_row <= max_tiles;
 }
 
+bool requires_padding_change(const ttnn::Tensor& tensor, ttnn::Layout layout) {
+    auto tile = tensor.get_tensor_spec().tile();
+    if (layout == Layout::ROW_MAJOR) {
+        // There shouldn't be extra paddings for Row Major layout
+        return tensor.logical_shape() != tensor.padded_shape();
```

Comment:
ideally this should be asserted?
also, should we change padded_shape to aligned_shape to make it clear that it has nothing to do with ttnn.pad?

Reply:
We shouldn't. What this check says is that if we're converting to row major, we should strip the padding if it was present

Reply:
This op uses `ttnn::pad`, so the `padded_shape` is kinda related to `pad`. I think we should keep it as `padded_shape`, at least while we still have Shape/LegacyShape and Padding.